x/programs/runtime/import_program.go

```diff
@@ -28,12 +53,11 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 	return &ImportModule{
 		Name: "program",
 		HostFunctions: map[string]HostFunction{
-			"call_program": {FuelCost: callProgramCost, Function: Function[callProgramInput, RawBytes](func(callInfo *CallInfo, input callProgramInput) (RawBytes, error) {
+			"call_program": {FuelCost: callProgramCost, Function: Function[callProgramInput, Result[RawBytes, ErrorCode]](func(callInfo *CallInfo, input callProgramInput) (Result[RawBytes, ErrorCode], error) {
 				newInfo := *callInfo
 
-				// make sure there is enough fuel in current store to give to the new call
-				if callInfo.RemainingFuel() < input.Fuel {
-					return nil, errors.New("remaining fuel is less than requested fuel")
+				if err := callInfo.ConsumeFuel(input.Fuel); err != nil {
+					return Err[RawBytes, ErrorCode](OutOfFuel), nil
```

Comment:
It's so nice that we now only have one place where the fuel consumption is calculated